paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -169,41 +205,34 @@ internal abstract class BaseSheetViewModel<TransitionTargetType>(
     @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
     fun setStripeIntent(stripeIntent: StripeIntent?) {
         _stripeIntent.value = stripeIntent
-
-        if (stripeIntent != null && getSupportedPaymentMethods().isEmpty()) {
-            onFatal(
-                IllegalArgumentException(
-                    "None of the requested payment methods" +
-                        " (${stripeIntent.paymentMethodTypes})" +
-                        " match the supported payment types" +
-                        " (${SupportedPaymentMethod.values().toList()})"
-                )
-            )
-        }
-
-        if (stripeIntent is PaymentIntent) {
-            runCatching {
-                _amount.value =
-                    Amount(
-                        requireNotNull(stripeIntent.amount),
-                        requireNotNull(stripeIntent.currency)
-                    )
-            }.onFailure {
-                onFatal(
-                    IllegalStateException("PaymentIntent must contain amount and currency.")
-                )
-            }
-        }
     }
 
-    fun getSupportedPaymentMethods(): List<SupportedPaymentMethod> {
-        stripeIntent.value?.let { stripeIntent ->
-            return stripeIntent.paymentMethodTypes.mapNotNull {
+    private fun getSupportedPaymentMethods(stripeIntent: StripeIntent?): List<SupportedPaymentMethod> {
+        stripeIntent?.let { it ->
+            return it.paymentMethodTypes.mapNotNull {
                 SupportedPaymentMethod.fromCode(it)
             }.filterNot {
                 // AfterpayClearpay requires a shipping address, filter it out if not provided
-                it == SupportedPaymentMethod.AfterpayClearpay &&
-                    (stripeIntent as? PaymentIntent)?.shipping == null
+                val excludeAfterPay = it == SupportedPaymentMethod.AfterpayClearpay &&
+                    (it as? PaymentIntent)?.shipping == null
+                if (excludeAfterPay && BuildConfig.DEBUG) {
+                    logger.debug("AfterPay will not be shown. It requires that Shipping is included in the Payment or Setup Intent")
```

Comment:
Are there current standard we have in place for logging already?  Should this just use Log.w so that it is not impacted by the logger configuration?  Currently this is not controlled by the merchant when using PaymentSheet

Reply:
cc @porter-stripe

Reply:
In iOS we have no set standard so I'm guessing in Android we don't either? I put some ideas around how we could create a shared logging framework in iOS. I think many of the ideas could extend to Android: https://paper.dropbox.com/doc/Logging-in-PaymentSheet-iOS--BSUx3FXIUliHrNUqB1oPnaBzAg-jg0mO5KvcjEC1Q9P93mDb

Reply:
We can have a `Logger.prod()` which only logs warning +

Reply:
In this scenario I think it makes sense to only log in the debug builds, when a developer is likely looking at it.